diff -up lua-5.1.2/src/luaconf.h.template.in~ lua-5.1.2/src/luaconf.h.template.in
--- lua-5.1.2/src/luaconf.h.template.in~	2007-10-21 19:33:30.000000000 +0200
+++ lua-5.1.2/src/luaconf.h.template.in	2007-10-21 19:33:30.000000000 +0200
@@ -98,7 +98,12 @@
 
 /** autotools-patch **/
 #define LUA_LDIR	"%lua_datadir%/lua/5.1/"
-#define LUA_CDIR	"%lua_libdir%/lua/5.1/"
+#if defined __alpha__ || defined __ia64__ || defined __powerpc64__ || \
+    defined __s390x__ || defined __sparc64__ || defined __x86_64__
+#define LUA_CDIR	"/usr/lib64/lua/5.1/"
+#else
+#define LUA_CDIR	"/usr/lib/lua/5.1/"
+#endif
 /** autotools-patch **/
 
 #define LUA_PATH_DEFAULT  \
